btcec/signature.go

```diff
@@ -76,10 +70,28 @@ func (sig *Signature) Serialize() []byte {
 	return b
 }
 
-// Verify calls ecdsa.Verify to verify the signature of hash using the public
-// key.  It returns true if the signature is valid, false otherwise.
 func (sig *Signature) Verify(hash []byte, pubKey *PublicKey) bool {
-	return ecdsa.Verify(pubKey.ToECDSA(), hash, sig.R, sig.S)
+	zero := new(nfieldVal)
+	if sig.S.Cmp(zero) == 0 || sig.R.Cmp(zero) == 0 {
+		return false
+	}
+	c := pubKey.Curve
+	tmp := hashToInt(hash, c)
+	e := new(nfieldVal).SetByteSlice(tmp.Bytes())
```

Comment:
It would be ideal to avoid the need for `big.Int` here.  Something like a `hashToNfield(hash)`.